src/tests/java/picard/sam/markduplicates/SimpleMarkDuplicatesWithMateCigar.java

```diff
@@ -0,0 +1,211 @@
+                        if (!record.getReadPairedFlag() || record.getMateUnmappedFlag()) {
+                            ++metrics.UNPAIRED_READ_DUPLICATES;
+                        } else {
+                            ++metrics.READ_PAIR_DUPLICATES;// will need to be divided by 2 at the end
+                        }
+                        numDuplicates++;
+                    }
+                    
+                    // To track optical duplicates, store a set of locations for mapped pairs, first end only
+                    if (doOpticalDuplicateTracking &&
+                            record.getReadPairedFlag() &&
+                            !record.getReadUnmappedFlag() &&
+                            !record.getMateUnmappedFlag() &&
+                            !readEndsSeen.contains(record.getReadName())) {
+                        
+                        final ReadEndsForSimpleMarkDuplicatesWithMateCigar readEnd = new ReadEndsForSimpleMarkDuplicatesWithMateCigar();
+                        // set orientation for optical duplicates
+                        if (record.getFirstOfPairFlag()) {
+                            readEnd.orientationForOpticalDuplicates = ReadEnds.getOrientationByte(record.getReadNegativeStrandFlag(), record.getMateNegativeStrandFlag());
+                        } else {
+                            readEnd.orientationForOpticalDuplicates = ReadEnds.getOrientationByte(record.getMateNegativeStrandFlag(), record.getReadNegativeStrandFlag());
+                        }
+                        if (opticalDuplicateFinder.addLocationInformation(record.getReadName(), readEnd)) {
+                            if (null != record.getReadGroup()) {
+                                final short index = libraryIdGenerator.getLibraryId(record);
+                                readEnd.setLibraryId(index);
+                            }
+                        }
+                        readEnds.add(readEnd);
+                        readEndsSeen.add(record.getReadName());
+                    }
+                }
+
+                if (!this.REMOVE_DUPLICATES || !record.getDuplicateReadFlag()) {
+                    if (PROGRAM_RECORD_ID != null) {
+                        record.setAttribute(SAMTag.PG.name(), chainedPgIds.get(record.getStringAttribute(SAMTag.PG.name())));
+                    }
+                    out.addAlignment(record);
+                    progress.record(record);
+                }
+            }
+
+            // Track the optical duplicates
+            if (1 < readEnds.size()) {
+                AbstractMarkDuplicatesCommandLineProgram.trackOpticalDuplicates(readEnds, opticalDuplicateFinder, libraryIdGenerator);
+            }
+        }
+
+        // remember to close the inputs
+        iterator.close();
+
+        out.close();
+
+        if (this.READ_NAME_REGEX == null) {
+            log.warn("Skipped optical duplicate cluster discovery; library size estimation may be inaccurate!");
```

Comment:
This is slightly confusing - In the base program trackOpticalDuplicates is only called if READ_NAME_REGEX is not null, but in this version you are not checking that condition (like 187 does not seem to depend on it). Could you unify the conditions? Also, in general, is markDuploicates from the base MarkDuplicates class ever called?